pkg/driver/dra_hooks.go

```diff
@@ -75,11 +78,42 @@ func (np *NetworkDriver) PublishResources(ctx context.Context) {
 	}
 }
 
+func (np *NetworkDriver) publishResourcesPrometheusMetrics(devices []resourceapi.Device) {
+	rdmaCount := 0
+	for _, device := range devices {
+		if *device.Attributes[apis.AttrRDMA].BoolValue {
+			rdmaCount++
+		}
+	}
+	publishedDevicesTotal.WithLabelValues("rdma").Set(float64(rdmaCount))
+	publishedDevicesTotal.WithLabelValues("non_rdma").Set(float64(len(devices) - rdmaCount))
+}
+
 func (np *NetworkDriver) PrepareResourceClaims(ctx context.Context, claims []*resourceapi.ResourceClaim) (map[types.UID]kubeletplugin.PrepareResult, error) {
 	klog.V(2).Infof("PrepareResourceClaims is called: number of claims: %d", len(claims))
+	start := time.Now()
+	defer func() {
+		draPluginRequestsLatencySeconds.WithLabelValues(methodPrepareResourceClaims).Observe(time.Since(start).Seconds())
+	}()
+	result, err := np.prepareResourceClaims(ctx, claims)
+
+	// identify errors and log metrics
+	isError := false
+	for _, res := range result {
+		if res.Err != nil {
+			isError = true
+			break
+		}
+	}
+	if isError || err != nil {
+		draPluginRequestsTotal.WithLabelValues(methodPrepareResourceClaims, statusFailed).Inc()
```

Comment:
More of a question, but should we maybe do this on a per claim level? If we are returning errors per claim then maybe it makes sense to expose the errors per claim, especially because one link may be the one that fails consistently. I'll let @gauravkghildiyal and @aojea weigh in.

Reply:
In this case, the problem is that the internal function never returns an explicit error so we would never know if there's a failure. I think we should add the separate metric right now.

Reply:
> I think we should add the separate metric right now.

Sounds good to me